Neuro/Minigames/Solvers/CleanO2FilterSolver.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections;
+using UnityEngine;
+using Neuro.Cursor;
+using System.Linq;
+
+namespace Neuro.Minigames.Solvers;
+
+[MinigameSolver(typeof(LeafMinigame))]
+public class CleanO2FilterSolver : MinigameSolver<LeafMinigame>
+{
+    protected override IEnumerator CompleteMinigame(LeafMinigame minigame, NormalPlayerTask task)
+    {
+        // ValidArea is somewhere offscreen so easier to just move towards the arrows on the chute
+        Vector3 exit = minigame.Arrows[0].transform.position;
+        Collider2D lastLeaf = null;
+        while (!task.IsComplete)
+        {
+            Collider2D leaf = minigame.Leaves
+                .Where(l => l && l.enabled && l.gameObject.active && l != lastLeaf 
```

Comment:
`lastLeaft` - Captured variable is modified in the outer scope

https://www.jetbrains.com/help/rider/AccessToModifiedClosure.html

```suggestion
            Collider2D _lastLeafCopy = lastLeaf;
            Collider2D leaf = minigame.Leaves
                .Where(l => l && l.enabled && l.gameObject.active && l != _lastLeafCopy 
```